core/models/payload.go

```diff
@@ -295,3 +314,31 @@ func (r *ResponseDetails) ConvertToResponseDetailsView() v1.ResponseDetailsView
 
 	return v1.ResponseDetailsView{Status: r.Status, Body: body, Headers: r.Headers, EncodedBody: needsEncoding}
 }
+
+func (r *ResponseDetails) ConvertToV2ResponseDetailsView() v2.ResponseDetailsView {
+	needsEncoding := false
+
+	// Check headers for gzip
+	contentEncodingValues := r.Headers["Content-Encoding"]
+	if len(contentEncodingValues) > 0 {
+		needsEncoding = true
+	} else {
+		mimeType := http.DetectContentType([]byte(r.Body))
+		needsEncoding = true
+		for _, v := range supportedMimeTypes {
+			if strings.Contains(mimeType, v) {
+				needsEncoding = false
```

Comment:
This code block is a little unclear (it might be the variable name, or perhaps two variables needed - needsEncoding and canEncode ?)